app/services/invoice_renderer.rb

```diff
@@ -1,5 +1,6 @@
 class InvoiceRenderer
   def render_to_string(order)
+    renderer.instance_variable_set(:@order, order)
```

Comment:
I would very much prefer `renderer.order = order`. These things tend to end badly.

Reply:
Agreed! I tried various nicer fixes in this case, but that solution would require a massive rewrite of lots of other connected bits of code, which also got increasingly messy.

Reply:
The whole implementation of this `InvoiceRenderer` is not great.

Reply:
I just had another look at this, and confirmed we get broken specs with `renderer.order = order`. 

```
NoMethodError: undefined method `order='
```

I can't tell you how nasty this was to debug and fix :see_no_evil:

Reply:
fair enough. Getting the upgrade out << instance_variable_set. Could you please create an issue for it so this fact doesn't get lost?